### PR TITLE
Match heuristically on url-parameters

### PR DESCRIPTION
**Pull request by JannikArndt**
This adds another heuristic matching regex which matches `src`-urls of type `…/e/eo?_t=…&_m=…&_e=…`.

### Background

I noticed a tracker with the url

```
https://links.fastic.com/e/eo?_t=31910dbd8025465fe404433a6734a&amp;_m=b94c48d2eb405b63b36e8662cbe95&amp;_e=hdnmlkv3BrstNKMdcMQ2RyrWI7YMy…
```

The company behind it is too small to roll their own tracking implementation, but they at least host or CNAME their solution.

If you go to http://links.fastic.com/e/eo directly, it tells you 
image

This leads you to add `_m` and `_e`…
image

…until you arrive at `Can't find the project that sent this link`.

My assumption is that this regex would match every company using this framework, for example trello is already on the list:
https://github.com/apparition47/MailTrackerBlocker/blob/9c073c7b3bb37b4f82a1bf661ac9d34891f1c062/Source/MTBBlockedMessage.m#L375

### Disclaimer
As you can tell from the code, I am _not_ an Obj-C dev 😅 This _should_ work, but please feel free to make it proper 😄

### CODE REVIEW

**Comment by JannikArndt** [comment]
Note: I used https://regex101.com/r/LfutSs/1 to create/test the regex:
image

**Comment by apparition47** [comment]
Any idea whose solution this is?

```html
<img 
src="https://links.fastic.com/e/eo?_t=31910dbd8025465fe404433a6734a&amp;_m=b94c48d2eb405b63b36e8662cbe95&amp;_e=hdnmlkv3BrstNKMdcMQ2RyrWI7Y" 
style="border:0;width:1px;height:1px;border-width:0px!important;display:none!important;line-height:0!improtant;" 
width="1" 
height="1" />
```

I'll see if I can look into who the vendor is behind this particular pattern is but the tracker in your example looks like it would already be picked up by the existing generic rule. Rather than adding a separate generic rule to catch `/e/eo/_t[^>]+_m[^>]+_e`, I think it would make sense to try identify this vendor specifically then add them into the existing `getTrackerDict` list.

```objc
        @"TataDocomoBusiness": @[@"tatadocomobusiness.com/rts/"],
        @"TBA Tracker XYZ": @[@"/e/eo/_t[^>]+_m[^>]+_e"],  //<--
        @"Techgig": @[@"tj_mailer_opened_count_all.php"],

**Comment by JannikArndt** [comment]
>Any idea whose solution this is?

No, I didn't find anything. But since it's a German company, they are required to list this in their privacy policy. They haven't. They don't even mention AWS, where this URL points (IP address & response headers). That's… a violation of the GDPR. I wrote them an e-mail 😈 

>the tracker in your example looks like it would already be picked up by the existing generic rule

Yes, my approach would be to have multiple, overlapping generic rules in places (assuming they don't "cost" a lot in terms of CPU time). 
However, the solution you proposed seems a lot cleaner 😊 I updated my branch and categorised the tracker as "Fastic" for now. If/when I receive an answer or learn more about the solution, I'll update the name (or create a new PR if you want to merge this already).

**Comment by JannikArndt** [comment]
Uh, the domain for the e-mail-address is not registered 😄 I now filed an official complaint with the authorities.

File: Source/MTBBlockedMessage.m [hunk 1]
```diff
@@ -207,6 +207,7 @@ - (NSDictionary*)getTrackerDict {
             @"facebookdevelopers.com/trk",
             @"fb.com/trk",
         ],
+        @"Fastic": @[@"/e/eo/_t[^>]+_m[^>]+_e"],
```
Comment (apparition47) [inline Source/MTBBlockedMessage.m:210]:
```suggestion
        @"Fastic": @[@"/e/eo\\?_t=[^>]+_m=[^>]+_e="],
```

Correct me if I'm wrong but the URL in your example didn't have a trailing `/` at the end of `/e/eo`

```html
<img 
<!--                             ⬇️ -->
src="https://links.fastic.com/e/eo?_t=31910dbd8025465fe404433a6734a&_m=b94c48d2eb405b63b36e8662cbe95&_e=hdnmlkv3BrstNKMdcMQ2RyrWI7Y" 
style="border:0;width:1px;height:1px;border-width:0px!important;display:none!important;line-height:0!improtant;" 
width="1" height="1" />
```

**Review by apparition47 — CHANGES_REQUESTED** [review]
Great!

And, I'm envious that this possible. Lay down the law!

File: Source/MTBBlockedMessage.m (hunk 1 above)
Comment (JannikArndt) [reply Source/MTBBlockedMessage.m]: Haha, you're right 😅 I just copied [your proposal from above](https://github.com/apparition47/MailTrackerBlocker/pull/99#issuecomment-812974271) 😛 Fixed it now 👍

**Review by apparition47 — APPROVED** [review]
Thanks again! Eager to find out who is behind this one.

**Pull request merged**